src/editors/hooks.js

```diff
@@ -13,16 +15,31 @@ export const navigateTo = (destination) => {
   window.location.assign(destination);
 };
 
-export const navigateCallback = (destination) => () => {
+export const navigateCallback = ({
+  destination,
+  analyticsEvent,
+  analytics,
+}) => () => {
+  if (process.env.NODE_ENV !== 'development' && analyticsEvent && analytics) {
+    sendTrackEvent(analyticsEvent, analytics);
+  }
```

Comment:
This is where the analytics is being sent. Effectively, it will happen when we navigate away from _any_ editor modal if analyticsEvent and analytics are passed to the call.